hack/update-codegen.sh

```diff
@@ -35,20 +35,23 @@ trap clean_up EXIT
 function generate_group() {
   local GROUP_NAME=$1
   local VERSION=$2
-  local CLIENT_PKG=${SCRIPT_PACKAGE}/pkg/client
-  local LISTERS_PKG=${CLIENT_PKG}/listers_generated local INFORMERS_PKG=${CLIENT_PKG}/informers_generated
+  local CLIENT_PKG=k8s.io/cluster-registry/pkg/client
+  local LISTERS_PKG=${CLIENT_PKG}/listers_generated
+  local INFORMERS_PKG=${CLIENT_PKG}/informers_generated
   local APIS_PKG=k8s.io/cluster-registry/pkg/apis
```

Comment:
In that case, I actually want "cluster-registry". If REPO_DIRNAME happens to be something other than cluster-registry, the script will break.

I believe this is a side effect of the Kubernetes codegen tools depending on their locations too heavily. I want this script to be location agnostic, so I've updated it so that it generates into a temporary location that is like a GOPATH and then copies it back to the actual location in the repository. This might not be a useful goal considering the rest of the ecosystem.